mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -222,134 +280,423 @@ public void testSimpleAttachment3() throws MessagingException, IOException {
+        @SuppressWarnings("unchecked")
+        List<String> values = (List<String>)mail.getAttribute(StripAttachment.SAVED_ATTACHMENTS_ATTRIBUTE_KEY);
+        assertThat(values).hasSize(2);
+    }
 
-        Assert.assertFalse(mail.getMessage().getContent() instanceof MimeMultipart);
+    @Test
+    public void analyseMultipartPartMessageShouldPutTwoPartsInCustomAttributeWhenTwoPartsMatch() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
 
-        Assert.assertEquals("simple text", mail.getMessage().getContent());
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "matched")
+                .setProperty("directory", folderPath)
+                .setProperty("pattern", ".*")
+                .setProperty("attribute", "my.custom.attribute")
+                .build();
+        mailet.init(mci);
+        MimeMultipart multiPart = new MimeMultipart();
+        
+        MimeBodyPart part1 = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part1.setFileName("removeMe1.tmp");
+        MimeBodyPart part2 = new MimeBodyPart(new ByteArrayInputStream(new byte[0]));
+        part2.setFileName("removeMe2.tmp");
+        multiPart.addBodyPart(part1);
+        multiPart.addBodyPart(part2);
+        MimeMessage message = new MimeMessage(Session
+                .getDefaultInstance(new Properties()));
+        message.setContent(multiPart);
+        message.saveChanges();
+        Mail mail = FakeMail.builder().build();
+        //When
+        boolean actual = mailet.analyseMultipartPartMessage(message, mail);
+        //Then
+        assertThat(actual).isTrue();
+        @SuppressWarnings("unchecked")
+        Map<String, byte[]> values = (Map<String, byte[]>)mail.getAttribute("my.custom.attribute");
+        assertThat(values).hasSize(2);
+    }
 
-        // prova per caricare il mime message da input stream che altrimenti
-        // javamail si comporta differentemente.
-        String mimeSource = "Message-ID: <26194423.21197328775426.JavaMail.bago@bagovista>\r\nSubject: test\r\nMIME-Version: 1.0\r\nContent-Type: text/plain; charset=us-ascii\r\nContent-Transfer-Encoding: 7bit\r\n\r\nsimple text";
+    @Test
+    public void analyseMultipartPartMessageShouldReturnTrueWhenAtLeastOneSubMultipartShouldHaveBeenRemoved() throws Exception {
+        //Given
+        StripAttachment mailet = new StripAttachment();
 
-        MimeMessage mmNew = new MimeMessage(Session
-                .getDefaultInstance(new Properties()),
-                new ByteArrayInputStream(mimeSource.getBytes("UTF-8")));
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("remove", "all")
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        MimeMultipart mainMultiPart = new MimeMultipart();
```

Comment:
This is hard to review. 

 - Maybe you can organize code a bit better or have more method extraction
 - A comment explaining structure of the message might be nice